extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/InfinispanClientProducer.java

```diff
@@ -224,6 +224,15 @@ private ConfigurationBuilder builderFromProperties(String infinispanClientName,
                     infinispanClientRuntimeConfig.sslCiphers.get().stream().collect(Collectors.joining(" ")));
         }
 
+        if (infinispanClientRuntimeConfig.sslHostNameValidation.isPresent()) {
+            properties.put(ConfigurationProperties.SSL_HOSTNAME_VALIDATION,
+                    infinispanClientRuntimeConfig.sslHostNameValidation.get());
+        }
+
+        if (infinispanClientRuntimeConfig.sniHostName.isPresent()) {
```

Comment:
Here, you already know that `sniHostName` is not set, while hostname and validation is enabled. You could throw configuration exception. IMO it would be nicer than having it thrown during bean creating on startup event. But I take it you don't want to repeat validation that is already in the Infinispan client?

**It's really just FYI, nothing else.**

```
21:07:04,706 INFO  [app] 21:07:03,928 Failed to start application (with profile [prod]): java.lang.RuntimeException: Failed to start quarkus
21:07:04,706 INFO  [app] 	at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
21:07:04,706 INFO  [app] 	at io.quarkus.runtime.Application.start(Application.java:101)
21:07:04,707 INFO  [app] 	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:111)
21:07:04,707 INFO  [app] 	at io.quarkus.runtime.Quarkus.run(Quarkus.java:71)
21:07:04,707 INFO  [app] 	at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
21:07:04,707 INFO  [app] 	at io.quarkus.runtime.Quarkus.run(Quarkus.java:124)
21:07:04,707 INFO  [app] 	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
21:07:04,707 INFO  [app] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
21:07:04,707 INFO  [app] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
21:07:04,707 INFO  [app] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
21:07:04,707 INFO  [app] 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
21:07:04,708 INFO  [app] 	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:61)
21:07:04,708 INFO  [app] 	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:32)
21:07:04,708 INFO  [app] Caused by: jakarta.enterprise.inject.CreationException: Error creating synthetic bean [ddee68cd3f60c96290f55bc237588d70e5c96aab]: org.infinispan.commons.CacheConfigurationException: ISPN004112: The SNI hostname is required when hostname validation is enabled
21:07:04,708 INFO  [app] 	at org.infinispan.client.hotrod.RemoteCacheManager_ddee68cd3f60c96290f55bc237588d70e5c96aab_Synthetic_Bean.doCreate(Unknown Source)
21:07:04,708 INFO  [app] 	at org.infinispan.client.hotrod.RemoteCacheManager_ddee68cd3f60c96290f55bc237588d70e5c96aab_Synthetic_Bean.create(Unknown Source)
21:07:04,708 INFO  [app] 	at org.infinispan.client.hotrod.RemoteCacheManager_ddee68cd3f60c96290f55bc237588d70e5c96aab_Synthetic_Bean.create(Unknown Source)
21:07:04,708 INFO  [app] 	at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:113)
21:07:04,709 INFO  [app] 	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:37)
21:07:04,709 INFO  [app] 	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:34)
21:07:04,709 INFO  [app] 	at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:32)
21:07:04,709 INFO  [app] 	at io.quarkus.arc.impl.ComputingCache.computeIfAbsent(ComputingCache.java:69)
21:07:04,709 INFO  [app] 	at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:34)
21:07:04,709 INFO  [app] 	at io.quarkus.arc.impl.ClientProxies.getApplicationScopedDelegate(ClientProxies.java:21)
21:07:04,709 INFO  [app] 	at org.infinispan.client.hotrod.RemoteCacheManager_ddee68cd3f60c96290f55bc237588d70e5c96aab_Synthetic_ClientProxy.arc$delegate(Unknown Source)
21:07:04,709 INFO  [app] 	at org.infinispan.client.hotrod.RemoteCacheManager_ddee68cd3f60c96290f55bc237588d70e5c96aab_Synthetic_ClientProxy.administration(Unknown Source)
21:07:04,709 INFO  [app] 	at io.quarkus.ts.messaging.infinispan.grpc.kafka.InfinispanPopulated.onStart(InfinispanPopulated.java:28)
21:07:04,710 INFO  [app] 	at io.quarkus.ts.messaging.infinispan.grpc.kafka.InfinispanPopulated_Observer_onStart_acc4e7f81539047dcf8d1cef111350cdc8ea75f8.notify(Unknown Source)
21:07:04,710 INFO  [app] 	at io.quarkus.arc.impl.EventImpl$Notifier.notifyObservers(EventImpl.java:346)
21:07:04,710 INFO  [app] 	at io.quarkus.arc.impl.EventImpl$Notifier.notify(EventImpl.java:328)
21:07:04,710 INFO  [app] 	at io.quarkus.arc.impl.EventImpl.fire(EventImpl.java:82)
21:07:04,710 INFO  [app] 	at io.quarkus.arc.runtime.ArcRecorder.fireLifecycleEvent(ArcRecorder.java:155)
21:07:04,710 INFO  [app] 	at io.quarkus.arc.runtime.ArcRecorder.handleLifecycleEvents(ArcRecorder.java:106)
21:07:04,710 INFO  [app] 	at io.quarkus.deployment.steps.LifecycleEventsBuildStep$startupEvent1144526294.deploy_0(Unknown Source)
21:07:04,711 INFO  [app] 	at io.quarkus.deployment.steps.LifecycleEventsBuildStep$startupEvent1144526294.deploy(Unknown Source)
21:07:04,711 INFO  [app] 	... 13 more
21:07:04,711 INFO  [app] Caused by: org.infinispan.commons.CacheConfigurationException: ISPN004112: The SNI hostname is required when hostname validation is enabled
21:07:04,711 INFO  [app] 	at org.infinispan.client.hotrod.configuration.SslConfigurationBuilder.validate(SslConfigurationBuilder.java:260)
21:07:04,711 INFO  [app] 	at org.infinispan.client.hotrod.configuration.SecurityConfigurationBuilder.validate(SecurityConfigurationBuilder.java:51)
21:07:04,711 INFO  [app] 	at org.infinispan.client.hotrod.configuration.ConfigurationBuilder.validate(ConfigurationBuilder.java:596)
21:07:04,711 INFO  [app] 	at org.infinispan.client.hotrod.configuration.ConfigurationBuilder.build(ConfigurationBuilder.java:664)
21:07:04,711 INFO  [app] 	at org.infinispan.client.hotrod.configuration.ConfigurationBuilder.build(ConfigurationBuilder.java:659)
21:07:04,712 INFO  [app] 	at io.quarkus.infinispan.client.runtime.InfinispanClientProducer.initialize(InfinispanClientProducer.java:109)
21:07:04,712 INFO  [app] 	at io.quarkus.infinispan.client.runtime.InfinispanClientProducer.getNamedRemoteCacheManager(InfinispanClientProducer.java:428)
21:07:04,712 INFO  [app] 	at io.quarkus.infinispan.client.runtime.InfinispanRecorder$1.apply(InfinispanRecorder.java:36)
21:07:04,712 INFO  [app] 	at io.quarkus.infinispan.client.runtime.InfinispanRecorder$1.apply(InfinispanRecorder.java:33)
21:07:04,712 INFO  [app] 	at io.quarkus.infinispan.client.runtime.InfinispanRecorder$InfinispanClientSupplier.get(InfinispanRecorder.java:82)
21:07:04,712 INFO  [app] 	at io.quarkus.arc.runtime.ArcRecorder$4.apply(ArcRecorder.java:129)
21:07:04,712 INFO  [app] 	at io.quarkus.arc.runtime.ArcRecorder$4.apply(ArcRecorder.java:126)
21:07:04,712 INFO  [app] 	at org.infinispan.client.hotrod.RemoteCacheManager_ddee68cd3f60c96290f55bc237588d70e5c96aab_Synthetic_Bean.createSynthetic(Unknown Source)
21:07:04,713 INFO  [app] 	... 34 more
```

Reply:
This kind of validation depends on Infinispan, not in Quarkus logic. I won't duplicate that here, since this logic can change for security or any other reasons.

Reply:
> this logic can change for security or any other reasons in Infinispan.

But you document it, how can you document it in Quarkus and say it can change in Infinispan? It will leave Quarkus documentation inconsistent.

Reply:
Anyway, I think I get your point, thanks for your answer and this PR.